Add blog post for Nanduri et al. (#57)
blab/blotter: blog/_posts/2024-11-15-dimensionality-reduction-for-flu-and-sars-cov-2.md

---
layout: post
title: Automated maps of seasonal flu and SARS-CoV-2 viruses show important evolutionary groups
author: John Huddleston
date: 2024-11-15
---

<style>
figure {
    border: 1px solid #CCCCCC;
    padding: 1em;
    width: 80%;
}
</style>

### Why do we get sick from the flu or SARS-CoV-2 so many times in our lives?

As I write this, I'm getting over a week-long cold caused by some virus that probably wasn't SARS-CoV-2 (the only virus I can test for at home).
The odds are good that it was a type of virus like the seasonal flu that has infected me before and that has now managed to escape my existing immunity.
This kind of reinfection happens all of the time.
Viruses exist only because they succeed in accomplishing two main goals (Figure 1):

1. make more copies of themselves
2. transmit from one host to another

<figure>
    <img src="/images/blog/cartography-virus-goals.jpg" />
    <figcaption><strong>Figure 1. Viruses have two goals: make more copies and infect new hosts.</strong>. Each larger orange circle represents a single copy of a virus.</figcaption>
</figure>

> When a virus infects us, it makes many more copies of itself with a pretty terrible copier machine that makes mistakes or “mutations” with each copy.

Reviewer: _nit_

```suggestion
When a virus infects us, it makes many more copies of itself with a pretty terrible copy machine that makes mistakes or “mutations” with each copy.
```

The new mutated copies are still close enough to the original to be considered the same type of virus (like seasonal flu) but different enough that our immune systems may not recognize them.

When we sneeze or cough in an elevator and transmit one of those mutated copies to someone else, the copy could look different enough to that person's immune system that the virus can infect them again, make more copies of itself with more mutations, and then transmit again to someone new.
For a prettier visual explanation of this process, check out [Jonathan Corum's and Carl Zimmer's beautiful article about how coronavirus mutates and spreads](https://www.nytimes.com/interactive/2020/04/30/science/coronavirus-mutations.html).

### What can we learn about mutations we find in viruses?

> As a virus researcher in [Trevor Bedford's lab](https://bedford.io/) at the Fred Hutchinson Cancer Center, I spend a lot of time thinking about these mutations that viruses make.

Reviewer: _nit:_ "make" implies intentionality on the part of the virus.

```suggestion
As a virus researcher in [Trevor Bedford's lab](https://bedford.io/) at the Fred Hutchinson Cancer Center, I spend a lot of time thinking about these viral mutations.
```

> For example, when we find a lot of seasonal flu viruses with the same mutation that allows those viruses to reinfect a lot of people in world, we can usually track that mutation back to a single common ancestor of all those recently successful virus copies.

Reviewer: _nit_
```suggestion
For example, when we find a lot of seasonal flu viruses with the same mutation that allows those viruses to reinfect a lot of people in the world, we can usually track that mutation back to a single common ancestor of all those recently successful virus copies.
```

</figure>

Unfortunately, it is possible to get infected by multiple copies of the same type of virus at the same time.
When this infection by multiple copies happens, the different infecting virus copies can make new copies of themselves in the same place in our bodies and accidentally include bits of each other in the new copies.
These bigger changes in the new virus copies break the rules that allow us to make virus family trees and they happen often enough that researchers have spent a lot of time making new computational tools to make family trees for viruses that have multiple parents.

### Could we find groups of related flu viruses without building a family tree?

In the Bedford lab, we recently stumbled on a new approach to find groups of virus copies that share the same mutations no matter how many parents they have and without building a family tree at all.

> This approach was a long time in the making, though, and started in July 2019 when a rising junior in high school, [Sravani Nanduri](https://bedford.io/team/sravani-nanduri/), joined the lab for a 2-month summer internship under joint mentorship of [Alli Black](https://bedford.io/team/allison-black/) and me.

Reviewer: _nit_

```suggestion
This approach was a long time in the making, though, and started in July 2019 when a rising junior in high school, [Sravani Nanduri](https://bedford.io/team/sravani-nanduri/), joined the lab for a 2-month summer internship under the joint mentorship of [Alli Black](https://bedford.io/team/allison-black/) and myself.
```

Sravani already knew how to write computer programs, but she wanted to learn more about programming and data visualization for biology.

Her internship project came from an idea Trevor had: what if, instead of building family trees of viruses based on their shared mutations, we could put viruses on a two-dimensional map where the distances between each pair of viruses reflected the mutations that differed between them?

> We had a lot of questions for a 2-month internship project: How would we build these maps? Would same groups of viruses we see in a tree place together in the maps? What would the distance between any two virus copies actually _mean_ on one of these maps? How would we visualize these maps? What would be the most fun bits of this project for Sravani to work on?

Reviewer: _nit_

```suggestion
We had a lot of questions for a 2-month internship project: How would we build these maps? Would the same groups of viruses we see in a tree place together in the maps? What would the distance between any two virus copies actually _mean_ on one of these maps? How would we visualize these maps? What would be the most fun bits of this project for Sravani to work on?
```

> Sravani, Alli, Trevor, and I ended up sketching the following example of what a final visualization would be for the project (Figure 3), with the idea that Sravani would apply a couple of well-known methods to one type of virus and plot the resulting maps for each method along side the tree of the same virus copies.

Reviewer: _nit_

```suggestion
Sravani, Alli, Trevor, and I ended up sketching out the following example of what a final visualization would be for the project (Figure 3), with the idea that Sravani would apply a couple of well-known methods to one type of virus and plot the resulting maps for each method alongside the tree of the same virus copies.
```

  - plot trees and maps in a single interactive figure that allowed us to highlight bits of the tree or a map and see the same viruses in the other parts of the figure

By August 2019, Sravani had made the prototype below (Figure 4) from mutations in a type of seasonal flu called “H3N2” which causes the most hospitalizations and deaths each year.

<figure>
    <img src="/images/blog/cartography-final-internship-prototype.png" alt="Sravani's final internship prototype showing maps based on flu mutations" />
    <figcaption><strong>Figure 4. Static view of Sravani's final internship prototype showing individual viruses in a family tree (top left) and corresponding positions of the same viruses in maps based on flu mutations including PCA (top middle), MDS (top right), t-SNE (bottom left), and UMAP (bottom right).</strong> Viruses in two specific groups from the tree (blue and orange) have been selected to show how their placement in the tree compares to their placement in the maps.</figcaption>
</figure>

The prototype revealed some interesting patterns:

  1. Most of the maps placed pairs of viruses with the same mutations closer together than pairs with different mutations.

>   2. Some of the maps (like MDS's) actually acted like a real map with the distance between viruses on the map matching exactly the number of mutations that differed between those viruses.

Reviewer: ```suggestion
  2. Some of the mapping methods (like MDS) actually acted like a real map with the distance between viruses on the map matching exactly the number of mutations that differed between those viruses.
```

Author: I'm going to skip this suggestion only because it's the maps that act like real maps and not the methods that act like maps.

Reviewer: Totally fair — this is the point where I confess that my real motivation here was trying to find a phrasing that allowed the removal of the greengrocer's apostrophe's (sic) in the parenthetical map names... but maybe if you're considering that map a possession of the mapping technique, this isn't actually incorrect 🤔

Author: That makes sense. I'm ok with saying that a map belongs to a method as in "MDS's map".

>   3. Other maps (like t-SNE's) didn't act like real maps, but they tightly clustered similar viruses into groups in the same space where we could easily find those groups by eye.

Reviewer: ```suggestion
  3. Other mapping methods (like t-SNE) didn't act like real maps, but they tightly clustered similar viruses into groups in the same space where we could easily find those groups by eye.
```

  4. The groups of viruses in these maps often matched the groups we had already defined in the tree.

Sravani and I were excited enough about these results to agree that we should keep this project going a little longer.
In October 2019, we decided to meet once a month while Sravani refined the prototype above and drafted a short summary of the results in the form of a scientific paper that we could post online somewhere.

### Can we find groups of related flu viruses when we _can't_ build a family tree?

Sravani and I continued to meet monthly through the beginning of the SARS-CoV-2 pandemic, she learned how to write a scientific paper, wrote the first full draft of a paper, and referenced this work in her college applications.
By June 2023, we'd both been busy with other projects.
Sravani had been focused on class work as an undergraduate in the University of Washington's Computer Science program.
I had been working with the Nextstrain team on pandemic response efforts.

> By the summer of 2023, Sravani was eager to revise the original paper and publish it in a scientific journal.

Reviewer: I don't have a great suggestion for replacement, but the repeated "By {time period}," with just two intervening sentences, is clunky.

Author: I felt this, too, as I wrote it. I just replaced that "By the summer of 2023," to make sentence read:

> Despite our other commitments, Sravani was eager to revise the original paper and publish it in a scientific journal.

//var spec = "http://localhost:4000/images/blog/cartography-flu-ha-embeddings.minified.json";
vegaEmbed('#h3n2-ha-embeddings', spec).then(function(result) {
// Access the Vega view instance (https://vega.github.io/vega/docs/api/view/) as result.view
}).catch(console.error);
</script>

These results suggested that we could use these maps of viral mutations to automatically define new, meaningful groups of viruses that could be reviewed by experts instead of requiring experts to manually define these groups.
This result was surprising because the methods we use to make these maps have no understanding of virus evolution; they only have a sense of how many mutations are shared or not between pairs of viruses.

We also realized we could make maps from viruses that had multiple parents even when the standard methods to build family trees wouldn't work.
For example, each flu virus is made up of 8 separate pieces that need to get bundled together to make a complete virus.
When we get infected by a single flu virus, that virus will make copies of all 8 pieces and its child viruses will get those copies from the same parent.

> When we get infected by more than one flu virus at the same time, those viruses can accidentally swap some of their 8 pieces such that parts of their child viruses come from different parents.

Reviewer: ```suggestion
When we get infected by more than one flu virus at the same time, those viruses can accidentally swap some of their 8 pieces such that parts of their child viruses come from different parents. (Scientists call this process "reassortment".)
```
